workflow/rules/common.smk

```diff
@@ -38,6 +40,40 @@ elif os.path.exists("config.yaml"):
     configfile: "config.yaml"
 
 
+# NOTE the remote URL relies on snakemake sourcecache features (e.g.: infer_source_file) introduced in 6.8.1
+min_version("6.8.1")
+
+
+def cacheopen(fname, mode="r"):
+    """
+    get a file handle of the fname.
+    snakemake.sourcecache will automatically cache it on-the-fly
+    """
```

Comment:
These comments are not formatted properly. But we can fix this eventually when adding a docstring formatter to our linting step.